File: WorkoutApp/WorkoutApp/Models/MuscleGroup.swift
import Foundation

enum MuscleGroup: String, Codable, CaseIterable, Identifiable {
    case chest
    case back
    case lats
    case traps
    case shoulders
    case biceps
    case triceps
    case forearms
    case abs
    case obliques
    case lowerBack
    case glutes
    case quadriceps
    case hamstrings
    case calves
    case adductors
    case abductors
    case neck

    var id: String { rawValue }

    var displayName: String {
        switch self {
        case .lowerBack: return "Lower Back"
        default: return rawValue.capitalized
        }
    }
}

File: web/src/types/muscleGroup.ts
export const MUSCLE_GROUPS = [
  "chest",
  "back",
  "lats",
  "traps",
  "shoulders",
  "biceps",
  "triceps",
  "forearms",
  "abs",
  "obliques",
  "lowerBack",
  "glutes",
  "quadriceps",
  "hamstrings",
  "calves",
  "adductors",
  "abductors",
  "neck",
] as const;

export type MuscleGroup = (typeof MUSCLE_GROUPS)[number];

const DISPLAY_NAMES: Record<MuscleGroup, string> = {
  chest: "Chest",
  back: "Back",
  lats: "Lats",
  traps: "Traps",
  shoulders: "Shoulders",
  biceps: "Biceps",
  triceps: "Triceps",
  forearms: "Forearms",
  abs: "Abs",
  obliques: "Obliques",
  lowerBack: "Lower Back",
  glutes: "Glutes",
  quadriceps: "Quadriceps",
  hamstrings: "Hamstrings",
  calves: "Calves",
  adductors: "Adductors",
  abductors: "Abductors",
  neck: "Neck",
};

export function muscleDisplayName(muscle: MuscleGroup): string {
  return DISPLAY_NAMES[muscle];
}
